pkgs/by-name/vt/vtsls/package.nix

```diff
@@ -0,0 +1,92 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  nodejs_20,
+  git,
+  pnpm_8,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vtsls";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "yioneko";
+    repo = "vtsls";
+    rev = "server-v${finalAttrs.version}";
+    hash = "sha256-HCi9WLh4IEfhgkQNUVk6IGkQfYagg805Rix78zG6xt0=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    nodejs_20
+    git
```

Comment:
Why is git required?
In any case, I think here `gitMinimal` is enough.

Reply:
From what I've understood from vtsls build setup is that it mainly uses it to apply patches to the vscode project it pulls as a submodule. It does create some other files with git shas included, but I suspect those are just for debugging.

I could patch out the whole building part and apply those patches using nix, which I did look into. The patches are included in the vtsls repo but their paths are not relative to root. I just didn't know how to apply patches with nix from the repository it's building. And there's the path issue, which is probably solvable by substitution during prePatch hook.

I'll test out `gitMinimal`, thanks!